### PR TITLE
Use only checked math ops for Uint* types

### LINKED ISSUE #1218: Use checked functions for all uint operations
All types created from `common/src/uint/impls.rs` must use checked operations only, the same way as Amount does.

### PR DESCRIPTION
**Pull request by TheQuantumPhysicist**
Closes #1218

### CODE REVIEW

**Review by azarovh — APPROVED** [review]

File: common/src/chain/block/consensus_data.rs [hunk 1]
```diff
@@ -146,7 +146,7 @@ impl PoWData {
         let mut ret = !target;
         let mut ret1 = target;
         ret1.increment();
-        ret = ret / ret1;
+        ret = (ret / ret1)?;
```
Comment (iljakuklic) [inline common/src/chain/block/consensus_data.rs:149]: The `ìncrement` operation is still not checked.

File: consensus/src/pow/helpers.rs [hunk 2]
```diff
@@ -62,21 +64,22 @@ where
...
     let old_target = Uint256::try_from(old_target)
         .map_err(|_| ConsensusPoWError::PreviousBitsDecodingFailed(old_target))?;
 
     // new target is computed by  multiplying the old target by ratio of the actual timespan / target timespan.
     // see Bitcoin's Protocol rules of Difficulty change: https://en.bitcoin.it/wiki/Protocol_rules
-    let mut new_target = old_target * actual_timespan;
-    new_target = new_target / target_timespan;
+    let mut new_target = (old_target * actual_timespan).unwrap_or(Uint256::MAX);
+    new_target =
+        (new_target / target_timespan).expect("target_timespan cannot be zero as per its type");
```
Comment (iljakuklic) [inline consensus/src/pow/helpers.rs:82]: Not for this PR: This expect could be avoided if we had a `NonZero` version of the numeric types and an exceptionless division taking a `NonZeroBlah`.

**Review by iljakuklic — APPROVED** [review]
+1

File: common/src/chain/block/consensus_data.rs (hunk 1 above)
Comment (TheQuantumPhysicist) [reply common/src/chain/block/consensus_data.rs:149]: Maybe I should get rid of it altogether... Not sure how to fix this.

File: consensus/src/pow/helpers.rs (hunk 2 above)
Comment (TheQuantumPhysicist) [reply consensus/src/pow/helpers.rs:82]: It occurred to me yesterday while I was dancing with PoW target time, but that's a huge undertaking and I don't think it's worth it.

File: common/src/chain/block/consensus_data.rs (hunk 1 above)
Comment (TheQuantumPhysicist) [reply common/src/chain/block/consensus_data.rs]: I got rid of it and discovered a bug in PoW... thanks for pointing it out.

**Pull request merged**